ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -99,9 +105,10 @@
    * @param keySize The key size that this segment supports
    * @param valueSize The value size that this segment supports
    * @param config The store config used to initialize the index segment
+   * @param time the {@link Time} instance to use for last modified time
    */
   IndexSegment(String dataDir, Offset startOffset, StoreKeyFactory factory, int keySize, int valueSize,
-      StoreConfig config, StoreMetrics metrics) {
+      StoreConfig config, StoreMetrics metrics, Time time) {
```

Comment:
We need this for test purposes (we use Mock time) and especially for BlobStoreStats so that different index segments will have different last Modified time in secs